### PR TITLE
javascript patching function fix

### PR DESCRIPTION
**Pull request by asachin96**
The patching function has a bug where "fn" is used inside a function instead of "name".

### CODE REVIEW

File: test/JavaScriptTests/patchConsoleObjectTests.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+var dConsoleInfo = "";
+var consoleDebug = "";
+var consoleError = "";
+var consoleWarn = "";
+var consoleLog = "";
+var consoleInfo = "";
+
+var consoleDebugger = {
+    log(msg){
+        dConsoleLog += msg;
```
Comment (akrantz) [inline test/JavaScriptTests/patchConsoleObjectTests.js]: I was thinking there would be one "consoleOutput" string which would have a message with a newline so you can tell which output function was called, e.g. consoleDebugger.log() would do:
```
log(msg) {
   consoleOutput += "DEBUGGER LOG: " + msg + "\n";
}
```
so other messages would start with "DEBUGGING INFO: " or "CONSOLE LOG: " etc.

Each test would reset consoleOutput to an empty string, perform the output, then compare the consoleOuput string to the expected.

File: test/JavaScriptTests/patchConsoleObjectTests.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+};
+
+describe("patchConsoleObjectTests", function() {
+    var global = {};
+    beforeEach(function() {
+        global = {};
+        consoleOutput = "";
+    });
+    it("case with empty console object", function() {
+        patchConsoleObject(global, consoleObjectEmpty, consoleDebugger);
```
Comment (akroshg) [inline test/JavaScriptTests/patchConsoleObjectTests.js]: I didn't understand the point of these tests. Are these to test the console objects or the patch function logic?

Comment (asachin96) [reply test/JavaScriptTests/patchConsoleObjectTests.js]: These tests are for testing patch function logic.

Comment (akrantz) [reply test/JavaScriptTests/patchConsoleObjectTests.js]: @akroshg I wanted to be able to have tests to ensure the patching function works in the various cases. Let us know what you think.

File: test/JavaScriptTests/patchConsoleObjectTests.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        assert.strictEqual(Object.keys(global.console)[0], "noLog");
+    });
+    it("case where console object only has log method", function() {
+        patchConsoleObject(global, consoleLogOnly, consoleDebugger);
+        global.console.log("log test");
+        assert.strictEqual(consoleOutput.trim().split("\n").length, 2);
+        assert.strictEqual(consoleOutput.split("\n")[0], "CONSOLE LOG: log test");
+        assert.strictEqual(consoleOutput.split("\n")[1], "DEBUGGER LOG: log test");
+    });
+    it("case with compelete console object", function() {
```
Comment (akrantz) [inline test/JavaScriptTests/patchConsoleObjectTests.js]: misspelling: compelete => complete

File: test/JavaScriptTests/patchConsoleObjectTests.js (hunk 2 above)
Comment (asachin96) [reply test/JavaScriptTests/patchConsoleObjectTests.js]: @akroshg Can you please complete this PR. Let me know if there is anything to change.

File: bin/Debugger.Sample/Debugger.Sample.cpp [hunk 4]
```diff
@@ -354,7 +355,7 @@ JsErrorCode DefineHostCallback(
 JsErrorCode RedirectConsoleToDebugger(DebugProtocolHandler *handler)
 {
     // Get global.console object. If that is not defined just return, otherwise get the debugger.console and patch with global.console object
-
+    fprintf(stdout, "saasa");
```
Comment (akrantz) [inline bin/Debugger.Sample/Debugger.Sample.cpp]: Was this added for testing? It doesn't seem like this was meant to be checked in.

Comment (asachin96) [reply bin/Debugger.Sample/Debugger.Sample.cpp]: Sorry, removed it

**Pull request merged**